packages/jest-util/src/install_common_globals.js

```diff
@@ -12,45 +12,55 @@ import type {Global} from 'types/Global';
 
 function deepCopy(obj) {
   const newObj = {};
-  let value;
+
   for (const key in obj) {
-    value = obj[key];
-    if (typeof value === 'object' && value !== null) {
-      value = deepCopy(value);
+    if (obj.hasOwnProperty(key)) {
+      let value = obj[key];
+
+      if (typeof value === 'object' && value !== null) {
+        value = deepCopy(value);
+      }
+
+      newObj[key] = value;
     }
-    newObj[key] = value;
   }
+
   return newObj;
 }
 
-export default (global: Global, globals: ConfigGlobals) => {
-  // Forward some APIs
-  global.Buffer = Buffer;
-
-  // `global.process` is mutated by FakeTimers. Make a copy of the
-  // object for the jsdom environment to prevent memory leaks.
-  // Overwrite toString to make it look like the real process object
-  let toStringOverwrite;
-  if (Symbol && Symbol.toStringTag) {
-    // $FlowFixMe
-    toStringOverwrite = {
-      [Symbol.toStringTag]: 'process',
-    };
+export default (globalObject: Global, globals: ConfigGlobals) => {
+  let prototype = Object.getPrototypeOf(process);
+  const processObject = Object.create(prototype);
+  const processBlacklist = /^mainModule$/g;
+
+  // Sequentially execute all constructors over the object.
+  do {
+    // $FlowFixMe: constructor is always defined.
+    prototype.constructor.call(processObject);
+  } while ((prototype = Object.getPrototypeOf(prototype)));
+
+  // Forward some APIs.
+  globalObject.Buffer = global.Buffer;
+
+  // Make a copy of "process" to avoid memory leaks.
+  if (typeof Symbol !== 'undefined' && Symbol.toStringTag) {
```

Comment:
I think we can remove the if statement. We now only support versions of node that have symbol and we can assume Symbol.toStringTag is available.